.github/workflows/ci.yml

```diff
@@ -187,18 +187,45 @@ jobs:
     - name: Install
       run: |
         apt-get update -y
-        apt-get install -y ghc cabal-install
+        apt-get install -y autoconf build-essential zlib1g-dev libgmp-dev curl libncurses5 libtinfo5 libncurses5-dev libtinfo-dev
+        curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | BOOTSTRAP_HASKELL_NONINTERACTIVE=1 BOOTSTRAP_HASKELL_INSTALL_NO_STACK=1 sh
     - uses: actions/checkout@v1
-    # We need to break dependency circuit between optparse-applicative,
-    # process, unix and bytestring. Newer versions of Cabal are capable
-    # of coming up with these automatically, but older ones need a hint.
-    # Specifically, we need to inform Cabal that:
-    #
-    # * optparse-applicative doesn't use process;
-    # * QuickCheck uses the old random package; and
-    # * tasty doesn't use unix.
     - name: Test
       run: |
+        source ~/.ghcup/env
         cabal update
-        cabal install --dependencies-only --enable-tests --constraint 'optparse-applicative -process' --constraint 'QuickCheck +old-random' --constraint 'tasty -unix'
         cabal test
+
+  # We use github.com/haskell self-hosted runners for ARM testing.
+  # If they become unavailable in future, put ['armv7', 'aarch64']
+  # back to emulation jobs above.
+  arm:
+    needs: build
+    runs-on: [self-hosted, Linux, ARM64]
+    strategy:
+      fail-fast: true
+      matrix:
+        arch: [arm32v7, arm64v8]
+    steps:
+      - uses: docker://arm64v8/ubuntu:focal
```

Comment:
@hasufell I copied this incantation from your example, but could you advice what's the deal here? Why do we need an ARM container to run `find -delete`? Could it be the same container as below, in build steps?

Reply:
I'm also unclear on this. And we seem to use an arm64v8 container in both versions of the job. Is this right?

Reply:
That’s fine. I assume the only thing which matters is to run `rm` under docker to get correct access rights. Basically any container would do.

Reply:
Yeah, the problem is that the self hosted runners are just processes on the host and don't run in temporary environments.

Since we run docker, the resulting files will be owned by root instead of the github user and the next actions/checkout will fail (because it runs as github user).